py4cast/datasets/poesy/poesy_cli.py

```
@@ -7,56 +7,94 @@
+    compute_stats: bool = True,
+):
+    """
+    Prepares Poesy dataset for training.
+    This command will:
+        - create all needed folders
+        - computes statistics on all weather parameters.
+    """
+
     print("--> Preparing Poesy Dataset...")
 
     print("Load train dataset configuration...")
     with open(path_config, "r") as fp:
         conf = json.load(fp)
 
-    print("Computing stats on each parameter...")
-    conf["settings"]["standardize"] = True
-    train_ds, _, _ = dataset.from_json(
-        fname=path_config,
-        num_input_steps=2,
-        num_pred_steps_train=1,
-        num_pred_steps_val_test=1,
+    print("instantiate train dataset configuration...")
+    train_ds, _, _ = DatasetABC.from_dict(
+        PoesyAccessor,
+        name=path_config.stem,
+        conf=conf,
+        num_input_steps=num_input_steps,
+        num_pred_steps_train=num_pred_steps_train,
+        num_pred_steps_val_test=num_pred_steps_val_test,
     )
-    cds.compute_parameters_stats(dataset)
 
-    print("Computing time stats on each parameters, between 2 timesteps...")
-    conf["settings"]["standardize"] = True
-    train_ds, _, _ = DatasetABC.from_json(
-        fname=path_config,
-        num_input_steps=2,
-        num_pred_steps_train=1,
-        num_pred_steps_val_test=1,
-    )
-    cds.compute_time_step_stats(dataset)
+    print("Creating cache folder")
+    print(train_ds.cache_dir)
+    train_ds.cache_dir.mkdir(exist_ok=True)
+
+    if compute_stats:
+        print(f"Dataset stats will be saved in {train_ds.cache_dir}")
+
+        print("Computing stats on each parameter...")
+        train_ds.settings.standardize = False
+
+        cds.compute_parameters_stats(train_ds)
+
+        print("Computing time stats on each parameters, between 2 timesteps...")
+        train_ds.settings.standardize = True
+        try:
+            del train_ds.sample_list
+        except AttributeError:
+            pass
```

why do you need to delete this attribute ?
in any case, we should avoid silent fails like `try except pass`

---

just modified to avoid silent fails. The pb is that : `compute_time_step_stats` supposes that the dataset is already standardized. But just before this we acessed `sample_list`